agents/weather_agent/weather_monitor_agent.py

```diff
@@ -0,0 +1,189 @@
+
+        Args:
+            loops (int):
+                How many data points are to be collected per call. Defaults to 1.
+
+        """
+        time.sleep(2)
+        if params is None:
+            params = {}
+
+        loops = params.get('loops')
+        # If loops is None, use value passed to Agent init
+        if loops is None:
+            loops = self.loops
+
+        sleep_time = 2
+
+        with self.lock.acquire_timeout(0, job='acq') as acquired:
+            if not acquired:
+                self.log.warn("Could not start acq because {} is already running"
+                              .format(self.lock.job))
+                return False, "Could not acquire lock."
+
+            session.set_status('running')
+
+            self.take_data = True
+
+            while self.take_data:
+                data = {
+                    'timestamp': time.time(),
+                    'block_name': 'temps',
+                    'data': {}
+                }
+
+                for i in range(0, loops):
+                    data['data'] = self.module.weather_daq(loops)[i]
+                    self.agent.publish_to_feed('weather_data', data)
+                print("data published!")
+
+                time.sleep(sleep_time)
+
+            self.agent.feeds['weather_data'].flush_buffer()
+
+        return True, 'Acquisition exited cleanly.'
+
+    def stop_acq(self, session, params=None):
+        """
+        Stops acq process.
+        """
+        if self.take_data:
+            self.take_data = False
+            print('requested to stop taking data.')
+            self.module.interrupt_daq()
+            return True, 'data taking succesfully halted'
+        else:
+            return False, 'acq is not currently running'
+
+
+def make_parser(parser=None):
+    if parser is None:
+        parser = argparse.ArgumentParser()
+
+    pgroup = parser.add_argument_group('Agent Options')
+    pgroup.add_argument('--port', type=str,
+                        help="Path to USB node for the weather monitor")
+    pgroup.add_argument('--mode', type=str, choices=['idle', 'init', 'acq'],
+                        help="Starting action for the agent.")
+    pgroup.add_argument('--loops', type=int,
+                        help="How many data points to collect per data acquisition call")
+    return parser
+
+
+if __name__ == "__main__":
+    parser = make_parser()
+    args = site_config.parse_args(
+        agent_class='WeatherMonitorAgent', parser=parser)
+    startup = False
+
+    if args.mode == 'acq':
+        startup = True
+
+    agent, runner = ocs_agent.init_site_agent(args)
+
+    path = '/dev'
+    for fname in os.listdir(path):
+        if fname[0:6] == 'ttyUSB':
+            usb = os.path.join(path, fname)
+            path += usb
+
+    print(path)
+
+    wMonitor = WeatherMonitorAgent(agent, args.port, args.loops)
+    agent.register_task('init_weathermonitor', wMonitor.init_weather_monitor_task,
```

Comment:
can delete line 184 no? Because the code block for init_weather_monitor_task is deleted as per 

> You can delete this part of the docstring. I realize it's in the LS240 agent but I probably should remove it there too.